doc/classes/DisplayServer.xml

```diff
@@ -128,6 +128,32 @@
 				[b]Note:[/b] On macOS, sandboxed apps will save security-scoped bookmarks to retain access to the opened folders across multiple sessions. Use [method OS.get_granted_permissions] to get a list of saved bookmarks.
 			</description>
 		</method>
+		<method name="file_dialog_with_options_show">
+			<return type="int" enum="Error" />
+			<param index="0" name="title" type="String" />
+			<param index="1" name="current_directory" type="String" />
+			<param index="2" name="root" type="String" />
+			<param index="3" name="filename" type="String" />
+			<param index="4" name="show_hidden" type="bool" />
+			<param index="5" name="mode" type="int" enum="DisplayServer.FileDialogMode" />
+			<param index="6" name="filters" type="PackedStringArray" />
+			<param index="7" name="options" type="Array" />
+			<param index="8" name="callback" type="Callable" />
+			<description>
+				Displays OS native dialog for selecting files or directories in the file system with additional user selectable options.
+				[param options] is [Array] of [Dictionary]s with the following keys:
+				- [code]"name"[/code] - option names.
+				- [code]"values"[/code] - [PackedStringArray] of values.
+				- [code]"default"[/code] - default value index.
+				Callbacks have the following arguments: [code]bool status, PackedStringArray selected_paths, int selected_filter_index, Dictionary selected_option[/code].
```

Comment:
Could be more useful to list the arguments in GDScript format, i.e. `status: bool` etc.